utils/installer/install.sh

```diff
@@ -1,272 +1,271 @@
-    sudo emerge -avn dev-python/pynvim
-  else
-    pip3 install pynvim --user
+function install_pip_deps() {
+  echo "Verifying that pip is available.."
+  if ! python3 -m ensurepip &>/dev/null; then
+    print_missing_dep_msg "pip"
+    exit 1
   fi
+  echo "Installing with pip.."
+  for dep in "${__pip_deps[@]}"; do
+    pip3 install --user "$dep"
+  done
+  echo "all python dependencies are succesfully installed"
 }
 
-installpacker() {
-  git clone https://github.com/wbthomason/packer.nvim ~/.local/share/lunarvim/site/pack/packer/start/packer.nvim
+function backup_old_config() {
+  for dir in "${__lvim_dirs[@]}"; do
+    # we create an empty folder for subsequent commands \
+    # that require an existing directory
+    mkdir -p "$dir" "$dir.bak"
+    if command -v rsync &>/dev/null; then
+      rsync --archive -hh --partial --info=stats1 --info=progress2 \
+        --modify-window=1 "$dir" "$dir.bak"
+    else
+      cp -R "$dir/*" "$dir.bak/."
+    fi
+  done
+  echo "Backup operation complete"
 }
 
-cloneconfig() {
-  if [ -d "/data/data/com.termux" ]; then
-    sudo() {
-      eval "$@"
-    }
-    USER_BIN_DIR="$HOME/../usr/bin"
-  fi
-  echo "Cloning LunarVim configuration"
-  mkdir -p ~/.local/share/lunarvim
-  case "$@" in
-
-    *--testing*)
-      cp -r "$(pwd)" ~/.local/share/lunarvim/lvim
-      ;;
-    *)
-      git clone --branch "$LVBRANCH" https://github.com/lunarvim/lunarvim.git ~/.local/share/lunarvim/lvim
-      ;;
-  esac
-  mkdir -p "$HOME/.config/lvim"
-  sudo cp "$HOME/.local/share/lunarvim/lvim/utils/bin/lvim" "$USER_BIN_DIR"
-  sudo chmod a+rx "$USER_BIN_DIR"/lvim
-  cp "$HOME/.local/share/lunarvim/lvim/utils/installer/config.example-no-ts.lua" "$HOME/.config/lvim/config.lua"
-
-  nvim -u ~/.local/share/lunarvim/lvim/init.lua --cmd "set runtimepath+=~/.local/share/lunarvim/lvim" --headless \
-    +'autocmd User PackerComplete sleep 100m | qall' \
-    +PackerInstall
-
-  nvim -u ~/.local/share/lunarvim/lvim/init.lua --cmd "set runtimepath+=~/.local/share/lunarvim/lvim" --headless \
-    +'autocmd User PackerComplete sleep 100m | qall' \
-    +PackerSync
+function install_packer() {
+  git clone --progress --depth 1 https://github.com/wbthomason/packer.nvim \
+    "$LUNARVIM_RUNTIME_DIR/site/pack/packer/start/packer.nvim"
+}
 
-  printf "\nCompile Complete\n"
+function clone_lvim() {
+  echo "Cloning LunarVim configuration"
+  local CLONE_CMD="git clone --progress --branch $LVBRANCH \
+    --depth 1 https://github.com/$LV_REMOTE $LUNARVIM_RUNTIME_DIR/lvim"
+  printf "Running: %s" "$CLONE_CMD"
+  eval "$CLONE_CMD"
+}
 
-  if [ -e "$HOME/.local/share/lunarvim/lvim/init.lua" ]; then
-    echo 'config.lua already present'
-  else
-    cp "$HOME/.local/share/lunarvim/lvim/utils/installer/config.example.lua" "$HOME/.config/lvim/config.lua"
+function setup_shim() {
+  if [ ! -d "$INSTALL_PREFIX/bin" ]; then
+    mkdir -p "$INSTALL_PREFIX/bin"
   fi
+  cat >"$INSTALL_PREFIX/bin/lvim" <<EOF
```

Comment:
I would generally agree, but try to write that `sed` command in a way that doesn't expand variables. I hate it when I have a hard-coded expansion of `$HOME` in my scripts, and I would like to keep those variables. That's why I opted for this approach which is way easier to read/maintain, but it's also very possible that there's a nice way to do it with `sed`.

Some notes to keep in mind about making sure it's cross-platform:
- `sed -E` is different between Linux and Darwin
- `awk` isn't guaranteed to be available and/or work reliably on BSD

time to write it in `perl`? 😄

Reply:
I don't understand what you mean about the sed command expanding variables, that's the whole point.
Please explain to me how having two versions of the same file, `bin/lvim` and this one hidden in the setup script is more maintainable.

Reply:
We don't need to have `util/bin/lvim` as a static file anymore. And there's nothing to maintain that isn't handled by the installer.

Reply:
What am i missing here ?
https://github.com/LunarVim/LunarVim/pull/1052/files#diff-fe4492969d1077edb685fdb1cc8cf4e6a7d93dcf6eb20d6b4e8fdd908077a5fa

Reply:
I meant I'm gonna delete it in the next patch 😄

Reply:
Sorry to come back on the topic, but I still think this should not be generated from the script.
These lines are unnecessarily hidden in the setup script, what didn't you like about the solution I offered ?

Reply:
It's does not qualify as "hidden" because: 
1- It's not doing anything out of the ordinary, as `nvim -u` is a very common flag.
2- the binary is entirely dependent on the install script, and there is not a single useful thing that it does which is not controlled by the installer. 

This pattern is not as uncommon as you think it is, especially for a trivial shim. You can also see similar patterns in homebrew or nix for example. 

In a later version, the user will be prompted to add `lvim` as a shell alias/function, that will also help in maintaining autocompletion. or allow running as a default neovim config by adjusting some variables and adding a symlink to `vim.fn.stdpath "config"`

Reply:
I've given my arguments, if you don't want to do it, so be it